src/modules/trust-registry.js

```diff
@@ -0,0 +1,264 @@
+import { BTreeSet } from '@polkadot/types';
+import { typedHexDID } from '../utils/did';
+import { getDidNonce } from '../utils/misc';
+
+/**
+ * `Trust Registry` module.
+ */
+export default class TrustRegistryModule {
+  /**
+   * Creates a new instance of `StatusListCredentialModule` and sets the api
+   * @constructor
+   * @param {object} api - PolkadotJS API Reference
+   * @param signAndSend
+   */
+  constructor(api, signAndSend) {
+    this.api = api;
+    this.module = api.tx.trustRegistry;
+    this.signAndSend = signAndSend;
+  }
+
+  /**
+   * Initializes Trust Registry with the supplied parameters.
+   * @param convenerDid
+   * @param registryId
+   * @param name
+   * @param govFramework
+   * @param signingKeyRef
+   * @param nonce
+   * @param didModule
+   * @param params
+   * @param waitForFinalization
+   * @param params
+   * @returns {Promise<null>}
+   */
+  async init(
```

Comment:
This and the remaining methods in this call convert the actor DID to a typed one and get the nonce so the following lines are repeated in all methods
```
... = typedHexDID(..)
const lastNonce = nonce ?? (await getDidNonce(....
```

Better to move this to a separate function.